src/provider.toit

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2025 Toit contributors.
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file.
+
+import gpio
+import one-wire
+import sensors.providers
+
+import .ds18b20 show Ds18b20
+
+NAME ::= "toit.io/ds18b20"
+MAJOR ::= 1
+MINOR ::= 0
+
+class TemperatureSensor implements providers.TemperatureSensor-v1:
+  bus_/one-wire.Bus? := null
+  pin_/gpio.Pin? := ?
+  sensor_/Ds18b20? := ?
+
+  constructor pin/int --id/int? --pull-up/bool=false:
+    pin_ = gpio.Pin pin
+    if id:
+      bus_ = one-wire.Bus pin_ --pull-up=pull-up
+      sensor_ = Ds18b20 --id=id --bus=bus_
+    else:
+      // There is no way to read the ID through the service interface API.
```

Comment:
Are you trying to explain why `--skip-id-read` is passed?

Reply:
yes. I moved the comment to the flag.